mono/metadata/socket-io.c

```diff
@@ -1423,10 +1420,10 @@ ves_icall_System_Net_Sockets_Socket_Disconnect_internal (SOCKET sock, MonoBoolea
 	MONO_ENTER_GC_SAFE;
 
 	if (_wapi_disconnectex != NULL) {
-		if (!_wapi_disconnectex (sock, NULL, TF_REUSE_SOCKET, 0))
+		if (!_wapi_disconnectex (sock, NULL, reuse ? TF_REUSE_SOCKET : 0, 0))
```

Comment:
does this has to do with moving WSAGetLastError, or is it another fix? If it's another fix, please split into a different commit (same PR is fine)